### PR TITLE
[DSIP-19] Support zeppelin connections in the connection center, as well as external connections to the connection center in zeppelin tasks

### PR DESCRIPTION
**Pull request by xdu-chenrj**
Support zeppelin connections in the connection center, as well as external connections to the connection center in zeppelin tasks

issue: #14338

Related issues: [DSIP-19][Feature] Add connection center feature for DS https://github.com/apache/dolphinscheduler/issues/10283

Mail: https://lists.apache.org/thread/xl6pb3sbrt0ffrf1fltcph39s9w1pjlx

### CODE REVIEW

**Comment by xdu-chenrj** [comment]
Tested locally.

image

image

**Comment by xdu-chenrj** [comment]
@EricGao888 
I want to obtain the type parameter of the zeppelin task, but I am not sure how it is transmitted from the front-end. In order to quickly advance, I directly wrote a fixed value.

When building connection parameters in DataSourceUtils, it will be built based on the type of task.

```java
 DataSourceUtils.buildConnectionParams(DbType.valueOf("ZEPPELIN"), zeppelinTaskExecutionContext.getConnectionParams());
```

**Comment by Radeity** [comment]
Hi, @xdu-chenrj , thanks for your contribution  :D

Some suggestions for you before taking further look of this PR :

- You should better follow the PR template to describe this PR.
- Please add missing license header for `dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-zeppelin/pom.xml`.
- Please avoid using wild card import in codes, e.g. `import org.apache.dolphinscheduler.plugin.task.api.*;`.
- Please run `mvn spotless:apply` to format your code first.
- Please handle conflicts first.

**Comment by xdu-chenrj** [comment]
> Hi, @xdu-chenrj , thanks for your contribution :D
> 
> Some suggestions for you before taking further look of this PR :
> 
> * You should better follow the PR template to describe this PR.
> * Please add missing license header for `dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-zeppelin/pom.xml`.
> * Please avoid using wild card import in codes, e.g. `import org.apache.dolphinscheduler.plugin.task.api.*;`.
> * Please run `mvn spotless:apply` to format your code first.
> * Please handle conflicts first.



Okay, I got it

File: .asf.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #
 
 github:
-  description: Apache DolphinScheduler is the modern data orchestration platform. Agile to create high performance workflow with low-code
```
Comment (EricGao888) [inline .asf.yaml]: Seems not related to this PR?

File: deploy/kubernetes/dolphinscheduler/templates/_helpers.tpl [hunk 2]
```diff
@@ -177,38 +177,6 @@ Create a database environment variables.
       {{- end }}
 {{- end -}}
 
-{{/*
```
Comment (EricGao888) [inline deploy/kubernetes/dolphinscheduler/templates/_helpers.tpl]: What is this change for?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/DataSourceServiceImpl.java [hunk 3]
```diff
@@ -385,6 +385,15 @@ public Result<Object> checkConnection(DbType type, ConnectionParam connectionPar
             }
             return result;
         }
+        if (type == DbType.ZEPPELIN) {
```
Comment (EricGao888) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/DataSourceServiceImpl.java]: Zeppelin is not a db datasource. Is there a better way to define the type?

**Review by EricGao888 — COMMENTED** [review]
BTW, could u plz update related UT cases for Zeppelin tasks?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/DataSourceServiceImpl.java (hunk 3 above)
Comment (EricGao888) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/DataSourceServiceImpl.java]: If you plan to improve this later when merging two modules and refactoring related code, you should add a `TDOD` comment here at least : )

**Review by songjianet — DISMISSED** [review]

**Comment by xdu-chenrj** [comment]
>

> BTW, could u plz update related UT cases for Zeppelin tasks?

Okay, I got it

**Comment by xdu-chenrj** [comment]
> BTW, could u plz update related UT cases for Zeppelin tasks?

Okay, I got it

File: .asf.yaml (hunk 1 above)
Comment (xdu-chenrj) [reply .asf.yaml]: Yes, I didn't make any special changes to this. It's possible that I mistook it when synchronizing the latest code

File: deploy/kubernetes/dolphinscheduler/templates/_helpers.tpl (hunk 2 above)
Comment (xdu-chenrj) [reply deploy/kubernetes/dolphinscheduler/templates/_helpers.tpl]: The same goes for this one

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-zeppelin/src/test/java/org/apache/dolphinscheduler/plugin/datasource/zeppelin/ZeppelinDataSourceProcessorTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    @Test
+    void testTestConnection() throws Exception {
+        ZeppelinDataSourceParamDTO zeppelinDataSourceParamDTO =
+                (ZeppelinDataSourceParamDTO) zeppelinDataSourceProcessor.createDatasourceParamDTO(connectJson);
+        ZeppelinConnectionParam zeppelinConnectionParam =
+                zeppelinDataSourceProcessor.createConnectionParams(zeppelinDataSourceParamDTO);
+
+        Mockito.doNothing().when(zeppelinClient).login(zeppelinConnectionParam.getUsername(),
+                zeppelinConnectionParam.getPassword());
+        boolean isConnected = zeppelinDataSourceProcessor.testConnection(zeppelinConnectionParam);
```
Comment (EricGao888) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-zeppelin/src/test/java/org/apache/dolphinscheduler/plugin/datasource/zeppelin/ZeppelinDataSourceProcessorTest.java]: The reason why this case fails is that you explicitly use the `new` keyword to instantiate the `ZeppelinClient` in `ZeppelinDataSourceProcessor`. When you do the stubbing here, the mocked `zeppelinClient` object is not injected into the `zeppelinDataSourceProcessor` thus the stubbing takes no effect. The best practice is to decouple the process of instantiation of `ZeppelinClient` from `ZeppelinDataSourceProcessor`.
image

Comment (xdu-chenrj) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-zeppelin/src/test/java/org/apache/dolphinscheduler/plugin/datasource/zeppelin/ZeppelinDataSourceProcessorTest.java]: Thank you very much for your help

**Comment by EricGao888** [comment]
@xdu-chenrj Could please resolve the conflicts? Together with other PRs related to DSIP-19. Thanks : )

**Review by EricGao888 — APPROVED** [review]
Waiting for CI to pass

**Comment by EricGao888** [comment]
Thanks for helping improve zeppelin task plugin @xdu-chenrj
